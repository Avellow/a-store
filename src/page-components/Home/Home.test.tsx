import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
import { renderWithRouterAndProvider } from "../../tests/helpers/renderWithRouterAndProvider";

describe('Home component', () => {
    it('Home renders', () => {
        renderWithRouterAndProvider();
    });

    it('Home snapshot', () => {
        const home = renderWithRouterAndProvider();
        expect(home).toMatchSnapshot();
    });

    it('test made-in-alfa block link', async () => {
        renderWithRouterAndProvider();
        const madeInAlfaBlock = screen.getByTestId('made-in-alfa-block-link');
        userEvent.click(madeInAlfaBlock);
        expect(screen.getByTestId('made-in-alfa-page')).toBeInTheDocument();
    });

    it('test your-design block link', async () => {
        renderWithRouterAndProvider();
        const yourDesignBlock = screen.getByTestId('your-design-block-link');
        userEvent.click(yourDesignBlock);
        expect(screen.getByTestId('your-design-page')).toBeInTheDocument();
    });
});
