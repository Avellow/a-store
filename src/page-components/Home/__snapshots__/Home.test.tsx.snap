// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
      >
        <header
          class="header header"
        >
          <h1
            class="typography__component_h1nzy logoTitle typography__styrene-medium_k22x5 typography__bold_h1nzy"
          >
            <a
              href="/"
            >
              A-Store
            </a>
          </h1>
          <button
            class="button__component_onnkc button__ghost_onnkc button__m_onnkc button__component_1qjkj button__ghost_1qjkj button__withLeftAddons_onnkc"
            type="button"
          >
            <span
              class="button__addons_onnkc"
            >
              <svg
                fill="currentColor"
                focusable="false"
                height="35"
                role="img"
                viewBox="0 0 24 24"
                width="35"
              >
                <path
                  d="M3 13.25h18v-2.5H3v2.5zm0-7.5v2.5h18v-2.5H3zm0 12.5h18v-2.5H3v2.5z"
                />
              </svg>
            </span>
            <span
              class="button__text_onnkc"
            >
              <span
                class="menuText"
              >
                меню
              </span>
            </span>
          </button>
        </header>
        <main
          class="blocks page body"
        >
          <a
            class="linkBlock"
            data-testid="made-in-alfa-block-link"
            href="/made-in-alfa"
          >
            <div
              class="imgWrapper"
            >
              <img
                alt="ссылка"
                src="/img/frame46.jpeg"
              />
              <h2
                class="typography__component_h1nzy title typography__styrene-medium_qohs3 typography__bold_h1nzy typography__primary_1faea"
              >
                Сделано в Альфе
              </h2>
            </div>
          </a>
          <a
            class="linkBlock"
            data-testid="your-design-block-link"
            href="/your-design"
          >
            <div
              class="imgWrapper"
            >
              <img
                alt="ссылка"
                src="/img/frame45.jpeg"
              />
              <h2
                class="typography__component_h1nzy title typography__styrene-medium_qohs3 typography__bold_h1nzy typography__primary_1faea"
              >
                Свой дизайн
              </h2>
            </div>
          </a>
        </main>
        <footer
          class="footer footer"
        >
          <span
            class="rights typography__primary-small_xcdcz typography__medium_xcdcz"
          >
            © ООО "Альфа Фьюче Пипл", 
            2023
          </span>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
    >
      <header
        class="header header"
      >
        <h1
          class="typography__component_h1nzy logoTitle typography__styrene-medium_k22x5 typography__bold_h1nzy"
        >
          <a
            href="/"
          >
            A-Store
          </a>
        </h1>
        <button
          class="button__component_onnkc button__ghost_onnkc button__m_onnkc button__component_1qjkj button__ghost_1qjkj button__withLeftAddons_onnkc"
          type="button"
        >
          <span
            class="button__addons_onnkc"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="35"
              role="img"
              viewBox="0 0 24 24"
              width="35"
            >
              <path
                d="M3 13.25h18v-2.5H3v2.5zm0-7.5v2.5h18v-2.5H3zm0 12.5h18v-2.5H3v2.5z"
              />
            </svg>
          </span>
          <span
            class="button__text_onnkc"
          >
            <span
              class="menuText"
            >
              меню
            </span>
          </span>
        </button>
      </header>
      <main
        class="blocks page body"
      >
        <a
          class="linkBlock"
          data-testid="made-in-alfa-block-link"
          href="/made-in-alfa"
        >
          <div
            class="imgWrapper"
          >
            <img
              alt="ссылка"
              src="/img/frame46.jpeg"
            />
            <h2
              class="typography__component_h1nzy title typography__styrene-medium_qohs3 typography__bold_h1nzy typography__primary_1faea"
            >
              Сделано в Альфе
            </h2>
          </div>
        </a>
        <a
          class="linkBlock"
          data-testid="your-design-block-link"
          href="/your-design"
        >
          <div
            class="imgWrapper"
          >
            <img
              alt="ссылка"
              src="/img/frame45.jpeg"
            />
            <h2
              class="typography__component_h1nzy title typography__styrene-medium_qohs3 typography__bold_h1nzy typography__primary_1faea"
            >
              Свой дизайн
            </h2>
          </div>
        </a>
      </main>
      <footer
        class="footer footer"
      >
        <span
          class="rights typography__primary-small_xcdcz typography__medium_xcdcz"
        >
          © ООО "Альфа Фьюче Пипл", 
          2023
        </span>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
